File: modules/dms-vision/ios/DmsLog.swift
// The module's ONLY logging (privacy rule, README §1): static event codes, never a value, never
// frame content. __tests__/native-ios.test.ts forbids every other logging call in ios/.

import Foundation
import os

enum DmsLogCode: String {
  case sessionStarted
  case sessionStopped
  case sessionPaused
  case sessionResumed
  case cameraConfigFailed
  case cameraInterrupted
  case cameraRuntimeError
  case landmarkerFailed
  case gazeNetFailed
  case watchdogPaused
  case watchdogStopped
  case modelsReleased
  case thermalPaused
  case backgroundStopped
}

enum DmsLog {
  private static let log = OSLog(subsystem: "com.roadwise.dmsvision", category: "dms")

  static func code(_ c: DmsLogCode) {
    os_log("%{public}@", log: log, type: .info, c.rawValue)
  }
}

File: modules/dms-vision/android/src/main/java/expo/modules/dmsvision/DmsLog.kt
// The module's only logging: static event codes, never values (no times, landmarks, sizes or
// messages from the camera stack), so a log can never carry anything about the driver.

package expo.modules.dmsvision

import android.util.Log

object DmsLog {
  enum class Code {
    SESSION_STARTED, SESSION_STOPPED, SESSION_PAUSED, SESSION_RESUMED, BACKGROUND_STOPPED,
    THERMAL_PAUSED, WATCHDOG_PAUSED, WATCHDOG_STOPPED, MODELS_RELEASED, CAMERA_INTERRUPTED,
    CAMERA_ERROR, INTERRUPTION_ENDED, GAZE_NET_FAILED
  }

  private const val TAG = "DmsVision"

  fun code(c: Code) {
    Log.i(TAG, c.name)
  }
}
